src/mono/mono.proj

```diff
@@ -87,6 +89,77 @@
+          <PropertyGroup Condition="'$(NeedEmscriptenPackages)' == 'true'">
+            <NodePkgVersion>$(runtimewinx64MicrosoftNETCoreRuntimeWasmNodeTransportVersion)</NodePkgVersion>
+            <EmsdkCacheVersion>$(MicrosoftNETRuntimeEmscripten3134Cachewinx64Version)</EmsdkCacheVersion>
+            <EmsdkSdkVersion>$(MicrosoftNETRuntimeEmscripten3134Sdkwinx64Version)</EmsdkSdkVersion>
+          </PropertyGroup>
+        </When>
+        <When Condition="'$(BuildArchitecture)' == 'arm64'">
+          <PropertyGroup Condition="'$(NeedEmscriptenPackages)' == 'true'">
+            <NodePkgVersion>$(runtimewinarm64MicrosoftNETCoreRuntimeWasmNodeTransportVersion)</NodePkgVersion>
+            <EmsdkCacheVersion>$(MicrosoftNETRuntimeEmscripten3134Cachewinarm64Version)</EmsdkCacheVersion>
+            <EmsdkSdkVersion>$(MicrosoftNETRuntimeEmscripten3134Sdkwinarm64Version)</EmsdkSdkVersion>
+          </PropertyGroup>
+        </When>
+      </Choose>
+    </When>
+    <When Condition="'$(HostOS)' == 'osx'">
+      <Choose>
+        <When Condition="'$(BuildArchitecture)' == 'x64'">
+          <PropertyGroup Condition="'$(NeedEmscriptenPackages)' == 'true'">
+            <NodePkgVersion>$(runtimeosxx64MicrosoftNETCoreRuntimeWasmNodeTransportVersion)</NodePkgVersion>
+            <EmsdkCacheVersion>$(MicrosoftNETRuntimeEmscripten3134Cacheosxx64Version)</EmsdkCacheVersion>
+            <EmsdkSdkVersion>$(MicrosoftNETRuntimeEmscripten3134Sdkosxx64Version)</EmsdkSdkVersion>
+          </PropertyGroup>
+        </When>
+        <When Condition="'$(BuildArchitecture)' == 'arm64'">
+          <PropertyGroup Condition="'$(NeedEmscriptenPackages)' == 'true'">
+            <NodePkgVersion>$(runtimeosxarm64MicrosoftNETCoreRuntimeWasmNodeTransportVersion)</NodePkgVersion>
+            <EmsdkCacheVersion>$(MicrosoftNETRuntimeEmscripten3134Cacheosxarm64Version)</EmsdkCacheVersion>
+            <EmsdkSdkVersion>$(MicrosoftNETRuntimeEmscripten3134Sdkosxarm64Version)</EmsdkSdkVersion>
+          </PropertyGroup>
+        </When>
+      </Choose>
+    </When>
+  </Choose>
+
+  <ItemGroup Condition="'$(NeedEmscriptenPackages)' == 'true'">
+    <PackageReference Include="Microsoft.NET.Runtime.Emscripten.3.1.34.Cache.$(_portableHostOS)-$(BuildArchitecture)"
+                      PrivateAssets="all"
+                      Version="$(EmsdkCacheVersion)"
+                      PackageArch="$(BuildArchitecture)" />
+    <PackageReference Include="runtime.$(_portableHostOS)-$(BuildArchitecture).Microsoft.NETCore.Runtime.Wasm.Node.Transport"
+                      Version="$(NodePkgVersion)"
+                      PackageArch="$(BuildArchitecture)" />
+    <PackageReference Include="Microsoft.NET.Runtime.Emscripten.3.1.34.Sdk.$(_portableHostOS)-$(BuildArchitecture)"
+                      PrivateAssets="all"
+                      Version="$(EmsdkSdkVersion)"
+                      PackageArch="$(BuildArchitecture)" />
+  </ItemGroup>
```

Comment:
```suggestion
  <ItemGroup Condition="'$(NeedEmscriptenPackages)' == 'true'">
    <PackageReference Include="Microsoft.NET.Runtime.Emscripten.$(EmsdkVersion).Cache.$(_portableHostOS)-$(BuildArchitecture)"
                      PrivateAssets="all"
                      Version="$(EmsdkPackageVersion)"
                      PackageArch="$(BuildArchitecture)" />
    <PackageReference Include="runtime.$(_portableHostOS)-$(BuildArchitecture).Microsoft.NETCore.Runtime.Wasm.Node.Transport"
                      Version="$(NodePkgVersion)"
                      PackageArch="$(BuildArchitecture)" />
    <PackageReference Include="Microsoft.NET.Runtime.Emscripten.$(EmsdkVersion).Sdk.$(_portableHostOS)-$(BuildArchitecture)"
                      PrivateAssets="all"
                      Version="$(EmsdkPackageVersion)"
                      PackageArch="$(BuildArchitecture)" />
  </ItemGroup>
```

Reply:
There's not `EmsdkVersion` and `EmsdkPackageVersion` properties, so that would not work. 

We can try to improve the hardcoded 3.1.34 version in the package names in a followup PR.

Reply:
> There's not `EmsdkVersion` and `EmsdkPackageVersion` properties, so that would not work.
> 
> We can try to improve the hardcoded 3.1.34 version in the package names in a followup PR.

I mean in combination with https://github.com/dotnet/runtime/pull/100266#discussion_r1580090404